pkg/asset/machines/machineconfig/workload-partitioning.go

```diff
@@ -0,0 +1,127 @@
+
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/pkg/errors"
+)
+
+// CrioWorkloadDropinContents generates the content expected by Cri-O for workload partitioning
+//
+// Example output:
+// [crio.runtime.workloads.management]
+// label = "management.workload.openshift.io/cores"
+// annotation_prefix = "io.openshift.workload.management"
+// resources = { "cpu" = "", "cpuset" = "0-1", }
+func CrioWorkloadDropinContents(name, cpuset string) (string, error) {
+	/*
+		[crio.runtime.workloads.management]
+		label = "management.workload.openshift.io/cores"
+		annotation_prefix = "io.openshift.workload.management"
+		resources = { "cpu" = "", "cpuset" = "0-1", }
+	*/
+	crioIni := ini.Empty()
+	section := crioIni.Section(fmt.Sprintf("crio.runtime.workloads.%s", name))
+	err := section.ReflectFrom(&crioWorkloadCfg{
+		Label:            fmt.Sprintf(`"%s.workload.openshift.io/cores"`, name),
+		AnnotationPrefix: fmt.Sprintf(`"io.openshift.workload.%s"`, name),
+		Resources:        fmt.Sprintf(`{ "cpu" = "", "cpuset" = "%s", }`, cpuset),
+	})
+	if err != nil {
+		return "", errors.Wrap(err, "Could not reflect structure to INI")
+	}
+	crioBuf := new(bytes.Buffer)
+	_, err = crioIni.WriteTo(crioBuf)
+	if err != nil {
+		return "", errors.Wrap(err, "Could not write INI to buffer")
+	}
+	return crioBuf.String(), nil
+}
+
+type crioWorkloadCfg struct {
+	Label            string `ini:"label"`
+	AnnotationPrefix string `ini:"annotation_prefix"`
+	Resources        string `ini:"resources"`
+}
+
+// KubeletWorkloadDropinContents generates te content expected by Kubelet for workload partitioning
+//
+// Example output:
+// {
+//   "management": {
+//     "cpuset": "0-1"
+//   }
+// }
+func KubeletWorkloadDropinContents(name, cpuset string) (string, error) {
+	kubeletCfg, err := json.MarshalIndent(kubeletWorkload{
+		name: kubeletWorkloadEntry{
+			Cpuset: cpuset,
+		},
+	}, "", "  ")
+	if err != nil {
+		return "", errors.Wrap(err, "Could not marshall JSON")
+	}
+	return string(kubeletCfg), nil
+}
+
+type kubeletWorkloadEntry struct {
+	Cpuset string `json:"cpuset"`
+}
+type kubeletWorkload map[string]kubeletWorkloadEntry
+
+// ForWorkloadPartitioning creates the MachineConfig that configures Cri-O and
+// Kubelet with the workload partition from install-config.yaml
+// 'workloadSettings'
+func ForWorkloadPartitioning(name, cpuset, role string) (*mcfgv1.MachineConfig, error) {
+	crioCfg, err := CrioWorkloadDropinContents(name, cpuset)
+	if err != nil {
+		return nil, err
+	}
+	kubeletCfg, err := KubeletWorkloadDropinContents(name, cpuset)
+	if err != nil {
+		return nil, err
+	}
+	ignConfig := igntypes.Config{
+		Ignition: igntypes.Ignition{
+			Version: igntypes.MaxVersion.String(),
+		},
+		Storage: igntypes.Storage{
+			Files: []igntypes.File{
+				ignition.FileFromString(
+					fmt.Sprintf("/etc/crio/crio.conf.d/01-%s-workload", name),
+					"root", 0644, crioCfg),
+				ignition.FileFromString(
+					"/etc/kubernetes/workload-pinning",
```

Comment:
Does this filename not need specificity for the workload? Like `/etc/kubernetes/workload-management-pinning`.

Reply:
No, this is a single file which contains all the workloads.